### PR TITLE
Gather the errors and print logs before terminate the plugin.

### PR DESCRIPTION
**Pull request by daeho-ro**
Discussed in the issue #31, and this is how I pass the malformed csv files.
It may confuse to user why some tables are not created and should find the reason from the log file.

### CODE REVIEW

**Comment by cbruno10** [comment]
Hey @daeho-ro , thanks for raising this PR!

I like the idea of making the CSV plugin easier to use, but as you had stated in this PR, I do have concerns that it may result in users not understanding why certain CSV files were not loaded. I think that before adding this behaviour, I would really like the ability to send warning messages in the terminal as per https://github.com/turbot/steampipe/issues/1090.

@kaidaguerre I'm not sure the level of effort required, so would you be able to estimate how long a feature like this would take? I think it would be useful not only in this case, but also in other plugins where being able to send a message in the terminal would be much more preferable to throwing errors/failing like we do today.

**Comment by daeho-ro** [comment]
I think it is possible to give an option and then user can select the behavior how the plugin set the header. Maybe header option with the values always true or false and auto.

**Comment by daeho-ro** [comment]
I have added an option to skip those errors,  so user can select it. 

Furthermore, I have made another changes that is for the errors. Until now, when you met the error in the `tableCSV`, it is hard to find which CSV files are bad because the plugin only shows the error for a CSV. So, I collect the errors into a slice, and raise the error later which makes you can find all the errors from the log file.

Of course, the steampipe CLI can only shows an error and still need to be added the feature what you mention, but at least I want to add the logs for all bad CSV files.

**Review by cbruno10 — COMMENTED** [review]
@daeho-ro I'm currently reviewing https://github.com/turbot/steampipe-plugin-csv/pull/42, and based on what I can see in there, I'm wondering if we still need this PR as well, considering that the main errors we're looking to avoid here is around malformed headers (if I understand correctly). If you agree, then I propose we leave this PR in draft, and see how the plugins feels once https://github.com/turbot/steampipe-plugin-csv/pull/42 is merged in.

I did leave a few suggestions based on my initial review, but if we're choosing to leave this in draft, then please disregard them for now.

Thanks!

**Comment by cbruno10** [comment]
@daeho-ro I've merged in https://github.com/turbot/steampipe-plugin-csv/pull/42, so I'm not sure if we need this config argument and feature, as I believe most of the file loading errors people experience today are due to incompatible headers.

Are there other common cases you've encountered where you've had issues loading invalid CSVs?

File: csv/plugin.go [hunk 1]
```diff
@@ -37,29 +37,37 @@ const (
...
 			plugin.Logger(ctx).Error("csv.PluginTables", "create_table_error", err, "path", i)
-			return nil, err
+			errors = append(errors, err)
+		} else {
+			tables[base[0:len(base)-len(filepath.Ext(base))]] = table
 		}
 	}
 
+	if !(*csvConfig.SkipError) && len(errors) > 0 {
+		return nil, errors[0]
```
[the remark this answers is not in the document]
Comment (daeho-ro) [reply csv/plugin.go:62]: There can be many malformed csv files but the return should be the type of error. So just select the first one. 

If we return the error during the loop, we can find only one malformed csv and can fix it. After the second steampipe query, there can be another malformed csv. But if we throw the error after the loop, we can print the logs to the plugin-logs file for all of them, and return one of them to the steampipe cli.

**Comment by daeho-ro** [comment]
I have simplified the PR to be more convenient for debugging. There could be other cases such as the file is actually not a csv, or reading error and so on. This will help you to find which file is not suitable by running steampipe query just once.

File: csv/plugin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbruno10) [reply csv/plugin.go:62]: @daeho-ro Sorry for the late reply on this!

As a suggestion for better error handling, could we do the following:
- Loop through files and if any errors are encountered, log the message (like the code is currently doing) and store the current filename
- After the loop, if there are any files that fail to load, we can return an error with a custom message using `fmt.Errorf` with an output  like: `failed to load CSV files: file1.csv, file2.csv, file3.csv.gz. Please check plugin logs for errors.`

Please let me know your thoughts, thanks!

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@daeho-ro Sorry for the long delay in reviewing this, please check out my suggestion and let me know what you think, thanks!

**Pull request closed without merging**